torch/_export/graph_module.py

```diff
@@ -86,6 +185,67 @@ def recompile(self) -> torch.fx.graph_module.PythonCode:
         self._code = python_code.src
         return python_code
 
+    def __reduce__(self) -> Tuple[Callable[..., "ExportGraphModule"], Tuple[bytes]]:
+        """
+        Serialization of the ExportGraphModule. The FX serialization does not
+        serialize the underlying graph to preserve backwards-compatiblity and
+        instead retraces the graph module when loading.  This results in loss of
+        metadata that is later used for optimizations directly on the graph
+        module.  Since we want to preserve this metadata and we do not care that
+        much about BC, we will write our own serialization method.
+        """
+
+        def op_to_str(op):
+            # Replace the targets with their names since we cannot serialize the ops
+            if isinstance(op, torch._ops.HigherOrderOperator):
+                return f"torch.ops.{op.__name__}"
+            elif "torch" in op.__module__:
+                return str(op)
+            else:
+                warnings.warn(f"Could not convert op {op} to str")
+                return ""
+
+        gm_dict = self.__dict__.copy()
+        gm_dict["_graphmodule_cls_name"] = self.__class__.__name__
+
```

Comment:
what about GraphModule.meta (not the graph.nodes.meta field)

Reply:
It's also pickled (test case checks for in_spec/out_spec fields)